flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStoreTest.java

```diff
@@ -341,11 +350,45 @@ public void testMaximumCapacity() throws IOException {
     /** Tests that a session cluster can terminate gracefully when jobs are still running. */
     @Test
     public void testPutSuspendedJobOnClusterShutdown() throws Exception {
+        final Duration timeout = Duration.ofSeconds(5);
         try (final MiniCluster miniCluster =
                 new PersistingMiniCluster(new MiniClusterConfiguration.Builder().build())) {
             miniCluster.start();
-            final JobGraph jobGraph = JobGraphTestUtils.singleNoOpJobGraph();
+            final JobVertex vertex = new JobVertex("blockingVertex");
+            vertex.setInvokableClass(BlockingNoOpInvokable.class);
+            final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(vertex);
+            final JobID jobId = jobGraph.getJobID();
             miniCluster.submitJob(jobGraph);
+            CommonTestUtils.waitUntilCondition(
+                    () -> {
+                        try {
+                            // Wait until the job is marked as RUNNING
+                            if (miniCluster.getJobStatus(jobId).get() != JobStatus.RUNNING) {
```

Comment:
there are easier ways to go about this, like `JobMasterTriggerSavepointITCase.NoOpBlockingInvokable`.

Reply:
Good point, I have replaced the query logic with using a custom `NoOpBlockingInvokable`.